components/service/glean/src/main/java/mozilla/components/service/glean/Glean.kt

```diff
@@ -80,10 +93,17 @@ open class GleanInternalAPI internal constructor () {
             return
         }
 
+        // Monkey-patch the product-provided configuration with the data coming
+        // from the debug configuration, if available. This is required in addition to
+        // `debugConfiguration`'s setter in case the debug configuration got set before
+        // `configuration` was available.
+        this.configuration = if (debugConfiguration != null) {
+            configuration.setConfigOverride(debugConfiguration!!)
+        } else { configuration }
```

Comment:
Can this ever happen? The only situation I can imagine is when Glean was not initialized in the application object. But that seems like an error?